### PR TITLE
[Material] Add splash radius property to selection controls

### LINKED ISSUE #69196: Allow us to set the splash radius on selection controls
## Use case

Allow us to set the splash radius on selection controls (checkbox, radio, toggle).

## Proposal

Add a property as on IconButton to set the splashRadius:
https://api.flutter.dev/flutter/material/IconButton/splashRadius.html

### PR DESCRIPTION
**Pull request by perclasson**
## Description

With this change a you can set the splash radius of selection controls.

| Default Checkbox | With larger splash radius |
| --- | --- |
|  | |

| Default Radio | With larger splash radius |
| --- | --- |
|  | |

| Default Switch | With larger splash radius |
| --- | --- |
|  | |

## Related Issues

A follow up issue will be to add themes to the selection controls.

Closes https://github.com/flutter/flutter/issues/69196

## Tests

I added the following tests:

- Checkbox with splash radius set
- Radio with splash radius set
- Switch with splash radius set

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/toggleable.dart [hunk 1]
```diff
@@ -331,6 +330,17 @@ abstract class RenderToggleable extends RenderConstrainedBox {
     markNeedsPaint();
   }
 
+  /// The splash radius for the radial reaction.
+  double get splashRadius => _splashRadius;
+  double _splashRadius;
+  set splashRadius(double value) {
+    assert(value != null);
```
Comment (johnsonmh) [inline packages/flutter/lib/src/material/toggleable.dart:337]: I don't think this is necessary since `value` is non-nullable

**Review by johnsonmh — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/switch.dart [hunk 2]
```diff
@@ -229,6 +232,11 @@ class Switch extends StatefulWidget {
   /// The color for the button's [Material] when a pointer is hovering over it.
   final Color? hoverColor;
 
+  /// The splash radius.
```
Comment (JoseAlba) [inline packages/flutter/lib/src/material/switch.dart:235]: I don't know what we should update this too but I definitelly feel like we should change it

**Review by JoseAlba — APPROVED** [review]
Update comments.

File: packages/flutter/lib/src/material/checkbox.dart [hunk 3]
```diff
@@ -173,6 +174,11 @@ class Checkbox extends StatefulWidget {
   /// The color for the checkbox's [Material] when a pointer is hovering over it.
   final Color? hoverColor;
 
+  /// The splash radius.
```
[the remark this answers is not in the document]
Comment (perclasson) [reply packages/flutter/lib/src/material/checkbox.dart:177]: I can change it to:
`The splash radius of the circular [Material] ink response.`

File: packages/flutter/lib/src/material/switch.dart (hunk 2 above)
Comment (perclasson) [reply packages/flutter/lib/src/material/switch.dart:235]: Updated now.

File: packages/flutter/lib/src/material/radio.dart [hunk 4]
```diff
@@ -266,6 +267,11 @@ class Radio<T> extends StatefulWidget {
   /// The color for the radio's [Material] when a pointer is hovering over it.
   final Color? hoverColor;
 
+  /// The splash radius of the circular [Material] ink response.
+  ///
+  /// If null, default splash radius of [kRadialReactionRadius] is used.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/radio.dart:272]: If null, then [kRadialReactionRadius] is used.

Here and elsewhere.

File: packages/flutter/test/material/radio_test.dart [hunk 5]
```diff
@@ -356,6 +356,42 @@ void main() {
...
+              );
+            }),
+          ),
+        ),
+      );
+    }
+    await tester.pumpWidget(buildApp());
+    await tester.pumpAndSettle();
+    expect(
+      Material.of(tester.element(find.byKey(radioKey))),
```
Comment (HansMuller) [inline packages/flutter/test/material/radio_test.dart:390]: We could dispense with the key and just use find.byType(Radio)? Maybe the type parameter makes that problematic?

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/radio_test.dart (hunk 5 above)
Comment (perclasson) [reply packages/flutter/test/material/radio_test.dart:390]: As you say its the type parameter makes it problematic: https://github.com/dart-lang/sdk/issues/11923

I can change it to `find.byWidgetPredicate((Widget widget) => widget is Radio<int>)` to remove the key.

**Pull request merged**